### PR TITLE
Added More API's in MsalCPPoAuth2Cache

### PR DESCRIPTION
**Pull request by kreedula**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java [hunk 1]
```diff
@@ -862,7 +862,7 @@ private void setToCacheRecord(@NonNull final CacheRecord target,
     @Override
     public List<AccountRecord> getAccounts(@Nullable final String environment,
                                            @NonNull final String clientId) {
-        final String methodName = ":getAccounts";
+        final String methodName = ":getAllAccounts";
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: is this intentional?

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java [hunk 2]
```diff
@@ -148,4 +126,93 @@ public synchronized void saveCredentials(@NonNull final AccountRecord accountRec
...
+     *
+     * @param homeAccountId : HomeAccountId of the Account
+     * @param environment   : Environment of the Account
+     * @param realm         : Realm of the Account
+     * @return {@link AccountDeletionRecord}
+     */
+    public synchronized AccountDeletionRecord removeAccount(@NonNull final String homeAccountId,
+                                                            @NonNull final String environment,
+                                                            @NonNull final String realm) {
+        final List<Credential> credentials = getAccountCredentialCache().getCredentialsFilteredBy(
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java]: nit: do we need validateNonNull here?

Comment (kreedula) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java]: done

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java (hunk 1 above)
Comment (kreedula) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java]: good catch, Android Studio auto-rename might have refactored this, fixed

File: common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java [hunk 3]
```diff
@@ -148,4 +126,95 @@ public synchronized void saveCredentials(@NonNull final AccountRecord accountRec
...
+    public synchronized AccountDeletionRecord removeAccount(@NonNull final String homeAccountId,
+                                                            @Nullable final String environment,
+                                                            @NonNull final String realm) throws ClientException {
+        validateNonNull(homeAccountId, "homeAccountId");
+        validateNonNull(realm, "realm");
+        final List<Credential> credentials = getAccountCredentialCache().getCredentialsFilteredBy(
+                homeAccountId,
+                environment,
+                CredentialType.RefreshToken,
+                null, realm,
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/MsalCppOAuth2TokenCache.java]: Minor formatting weirdness here

**Pull request merged**